lib/rds/manager/Manager.rb

```diff
@@ -0,0 +1,188 @@
+        end
+
+        diffs.each do |diff|
+          case diff.type
+          when InstanceChange::ENGINE,
+               InstanceChange::USERNAME,
+               InstanceChange::SUBNET,
+               InstanceChange::DATABASE,
+               InstanceChange::STORAGE_TYPE
+            puts Colors.red("Cannot change #{diff.asset_type}")
+          when InstanceChange::PORT
+            puts "Updating the database port number..."
+            modify_instance.call({db_port_number: local.port})
+          when InstanceChange::TYPE
+            puts "Updating the database instance type..."
+            modify_instance.call({db_instance_class: "db." + local.type})
+          when InstanceChange::ENGINE_VERSION
+            puts "Updating the engine version..."
+            modify_instance.call({engine_version: local.engine_version})
+          when InstanceChange::STORAGE_SIZE
+            puts "Updating the allocated storage..."
+            modify_instance.call({allocated_storage: local.storage_size})
+          when InstanceChange::SECURITY_GROUPS
+            puts "Updating the security groups..."
+            security_group_ids = local.security_groups.map do |sg|
+              # TODO: once the db subnet config is created, find the security group id based on the vpc in that subnet.
+              # right now, if security groups from different vpc have the same name, this might not return the right id.
+              sg_id = SecurityGroups.vpc_security_group_id_names.values.inject(:merge).key(sg)
+              if sg_id.nil?
+                Exception.new("security group #{sg} does not exist")
+              end
+              sg_id
+            end
+            modify_instance.call({vpc_security_group_ids: security_group_ids})
+          when InstanceChange::PUBLIC
+            if local.public_facing
+              puts "making the database public..."
+            else
+              puts "blocking the database from the public..."
+            end
+            modify_instance.call({publicly_accessible: local.public_facing})
+          when InstanceChange::BACKUP
+            "Updating the backup preferences..."
+            modify_instance.call({preferred_backup_window: local.backup_window, backup_retention_period: local.backup_period})
+          when InstanceChange::UPGRADE
+            "Updating the upgrade preferences..."
+            modify_instance.call({preferred_maintenance_window: local.upgrade_window, auto_minor_version_upgrade: local.auto_upgrade})
+          end
+        end
+      end
+
+      def create(local)
+        errors = Array.new
+
+        if local.name.nil?
+          errors << "instance name is required"
+        end
+
+        if local.type.nil?
+          errors << "instance type is required"
+        end
+
+        if local.engine.nil?
+          errors << "database engine is required"
+        end
+
+        if !errors.empty?
+          puts "Could not create #{local.name}:"
```

Comment:
I blame whoever made the ec2 Manager which I used for reference, but ok. I'll make it red.